src/common.ts

```diff
@@ -105,27 +103,35 @@ export class ObjectToStructConverter {
   encodeValue_(value: {} | null): any {
     let convertedValue;
 
-    if (is.null(value)) {
+    if (value === null) {
       convertedValue = {
         nullValue: 0,
       };
-    } else if (is.number(value)) {
+    } else if (typeof value === 'number') {
       convertedValue = {
         numberValue: value,
       };
-    } else if (is.string(value)) {
+    } else if (typeof value === 'string') {
       convertedValue = {
         stringValue: value,
       };
-    } else if (is.boolean(value)) {
+    } else if (typeof value === 'boolean') {
       convertedValue = {
         boolValue: value,
       };
     } else if (Buffer.isBuffer(value)) {
       convertedValue = {
         blobValue: value,
       };
-    } else if (is.object(value)) {
+    } else if (Array.isArray(value)) {
+      convertedValue = {
+        listValue: {
+          values: (value as Array<{}>).map(this.encodeValue_.bind(this)),
+        },
+      };
+    } else if (value?.toString() === '[object Object]') {
+      // Using `typeof value === 'object'` is discouraged here, because it
```

Comment:
tricky!

Reply:
Yeah, it's like I was saying - `typeof foo === 'object'` is a weird footgun, so I'm conflicted on the change.  This check more accurately matches what `is.object` does in other places.